test/server/cards/01_SOR/events/SneakAttack.spec.ts

```diff
@@ -0,0 +1,30 @@
+describe('Sneak Attack', function() {
+    integration(function(contextRef) {
+        it('Sneak Attack ability\'s should play Sabine for 1 resource less and defeat it at the end.', function () {
+            contextRef.setupTest({
+                phase: 'action',
+                player1: {
+                    hand: ['sneak-attack', 'sabine-wren#you-can-count-on-me', 'obiwan-kenobi#following-fate'],
+                    groundArena: ['battlefield-marine'],
+                    spaceArena: ['cartel-spacer'],
+                    base: { card: 'administrators-tower', damage: 0 },
+                    leader: { card: 'luke-skywalker#faithful-friend', deployed: false },
+                    resources: 3
+                },
+            });
+
+            const { context } = contextRef;
+
+            context.player1.clickCard(context.sneakAttack);
+            expect(context.player1).toBeAbleToSelectExactly([context.sabineWren]);
+            context.player1.clickCard(context.sabineWren);
+            expect(context.sabineWren.exhausted).toBeFalse();
+            expect(context.player1.readyResourceCount).toBe(0);
+
+            // Check that Sabine is defeated at the beginning of the regroup phase
+            context.moveToRegroupPhase();
+            expect(context.sabineWren).toBeInZone('discard');
+            expect(context.player1).toHavePrompt('Select between 0 and 1 cards to resource');
+        });
```

Comment:
Delayed effects are new so it would be good to have an extra test case to validate something - can we have a test where the unit gets defeated before the end of the phase just to confirm nothing breaks?

Can be a separate test case or can just extend the existing one by moving the card back into hand and playing a different unit. I'm fine either way since moving the card back into hand is a manual action and we try to avoid those, but also reducing test cases is good. So whichever you prefer